daphne/src/testing.rs

```diff
@@ -470,32 +474,156 @@ impl DapLeader<BearerToken> for MockAggregator {
         Ok(HashMap::from([(selector.task_id.clone(), reports)]))
     }
 
-    async fn init_collect_job(&self, _collect_req: &CollectReq) -> Result<Url, DapError> {
-        unreachable!("not implemented");
+    // Called after receiving a CollectReq from Collector.
+    async fn init_collect_job(&self, collect_req: &CollectReq) -> Result<Url, DapError> {
+        let mut rng = thread_rng();
+        let task_config = self
+            .get_task_config_for(&collect_req.task_id)
+            .ok_or_else(|| DapError::fatal("no task found"))?;
```

Comment:
A bit clearer:
```suggestion
            .ok_or_else(|| DapError::fatal("task not found"))?;
```

Reply:
Sounds good. I've updated the error messages.